src/lsm/manifest_level.zig

```diff
@@ -54,6 +54,101 @@ pub fn ManifestLevelType(
             tables: std.BoundedArray(TableInfoReference, constants.lsm_growth_factor),
         };
 
+        const LevelKeyRange = struct {
+            key_min: ?Key = null,
+            key_max: ?Key = null,
+
+            fn get(range: *const LevelKeyRange) ?KeyRange {
+                assert(range.empty() or range.non_empty());
```

Comment:
This is in the hot path. So gate with `verify`.

Reply:
Also, we would normally use `maybe` instead to indicate that the range can be empty or not.

Reply:
Using a `key_range: ?KeyRange` in `LevelKeyRange` simplified the method such that we don't need the assert anymore! So that worked out nicely for the hot path. :) 
```C
            fn get(self: *const LevelKeyRange) ?KeyRange {
                return self.key_range;
            }
```